specs/corla.pvs

```diff
@@ -0,0 +1,923 @@
+    IMPLIES c1 = c2)
+  
+  election_canvass: TYPE
+  political_party: TYPE
+  audit_center: TYPE
+END elections
+
+% @relation state_election inherit election
+election: DATATYPE WITH SUBTYPES state_election
+BEGIN
+  IMPORTING elections
+  
+  is_state_election?(s: state, c: set[contest]): make_state_election: state_election
+END election
+
+%+END elections
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%+BEGIN ballots
+
+%%% \section{Ballots}
+
+%%% The second basic concept of elections that we need to specify is
+%%% ballots. Ballots are the means by which voters indicate their
+%%% preferences for candidates in contests and choices in ballot
+%%% questions.
+
+ballots: THEORY
+BEGIN
+  IMPORTING elections
+
+%%% Several of our concepts are simply numbers. For example, ballot
+%%% identifiers (ids), batch numbers, and batch sizes are all just
+%%% natural numbers (integers starting from zero).
+
+  ballot_id: TYPE = nat
+  batch_number: TYPE = nat
+  batch_size: TYPE = nat
+
+%%% Ballots are marked by voters or ballot marking devices.  A ballot
+%%% mark is any kind of mark made on a ballot that is not found on a
+%%% blank ballot. A voter_marking is any mark made by a voter. An
+%%% ambiguous mark is a mark for which there is ambiguity in its
+%%% interpretation (voter or machine). A stray mark is a mark that is
+%%% outside of the legitimate regions of a ballot, such as hesitation
+%%% marks outside of the mark regions for a contest or marks made by
+%%% coffee spilled on a paper ballot.
+
+  ballot_mark: TYPE
+  voter_marking: TYPE FROM ballot_mark
+  ambiguous_mark: TYPE FROM ballot_mark
+  stray_mark: TYPE FROM ballot_mark
+
+%%% Ballots come in several varieties.  All ballots are either
+%%% represented in a digital fashion (a scan, PDF, etc.) or are paper
+%%% ballots, or both. Ballots are also classifed based upon where they
+%%% originate, such as ballots delivered via mail, UOCAVA ballots,
+%%% early ballots, etc. Ballots are also either tabulated or not. Note
+%%% that none of these categories of ballots are mutually exclusive.
+
+% @description The legally recognized means by which a voter can
+% express his or her choices in one or more contests in an election.
+  ballot: TYPE
+  digital_ballot: TYPE FROM ballot
+  paper_ballot: TYPE FROM ballot
+  mail_ballot: TYPE FROM ballot
+  uocava_ballot: TYPE FROM ballot
+  early_ballot: TYPE FROM ballot
+  tabulated_ballot: TYPE FROM ballot
+
+%%% Each ballot has a single ballot style, which indicates (at least)
+%%% the contests that are listed on that ballot. Usually ballot styles
+%%% are encoded through the use of natural numbers.
+
+% @description The concrete presentation of a ballot configuration.
+  ballot_style: TYPE = nat
+  ballot_style?: [ballot -> ballot_style]
+
+% @description A decision to be made, either the selection of one or
+% more candidates for an office or the choice of a course of action.
+  ballot_contest: TYPE = contest
```

Comment:
Is it necessary here to indicate that a ballot potentially has multiple contests on it?